core/src/main/java/org/hisp/dhis/android/persistence/dataelement/DataElementDB.kt

```diff
@@ -57,4 +65,50 @@ internal data class DataElementDB(
     val fieldMask: String?,
     val color: String?,
     val icon: String?,
-)
+) : EntityDB<DataElement>, BaseNameableObjectDB {
+
+    override fun toDomain(): DataElement {
+        return DataElement.builder().apply {
+            applyBaseNameableFields(this@DataElementDB)
+            id(id?.toLong())
+            valueType?.let { valueType(ValueType.valueOf(it)) }
+            zeroIsSignificant(zeroIsSignificant)
+            aggregationType(aggregationType)
+            formName(formName)
+            domainType(domainType)
+            displayFormName(displayFormName)
+            optionSet?.let { optionSet(ObjectWithUidDB(it).toDomain()) }
```

Comment:
In Network we use sometimes the domain object directly, such as `ObjectWithUid.create(it)`. I don't have a preference, just wanted to mention because it wouldn't break the layer separation

Reply:
Same, I thought on it, don't have a clear preference. I selected this one so it is consistent with other more complex objects. Would you suggest to avoid creating more objects when is not needed and use the domain object directly?